core/orch_test.go

```diff
@@ -219,6 +220,47 @@ func TestSelectTranscoder(t *testing.T) {
 	assert.Len(n.TranscoderManager.remoteTranscoders, 7)
 }
 
+func TestTranscoderManagerTranscoding(t *testing.T) {
+	n, _ := NewLivepeerNode(nil, "", nil)
+	m := NewRemoteTranscoderManager()
+	s := &StubTranscoderServer{node: n}
+	r := NewRemoteTranscoder(n, s, 5)
+
+	// sanity checks
+	assert := assert.New(t)
+	assert.Empty(m.liveTranscoders)
+	assert.Empty(m.remoteTranscoders)
+
+	m.Register(r)
+	assert.Len(m.remoteTranscoders, 5) // sanity
+	assert.Len(m.liveTranscoders, 1)
+	assert.NotNil(m.liveTranscoders[s])
+
+	// happy path
+	res, err := m.Transcode("", nil)
+	assert.Nil(err)
+	assert.Len(res, 1)
+	assert.Equal(string(res[0]), "asdf")
+
+	// non-fatal error should not remove from list
+	s.TranscodeError = fmt.Errorf("TranscodeError")
```

Comment:
So here, when we do successfully receive results from remote transcoder, but there's a transcodeError, that doesn't warrant us stopping working with the transcoder? Is it because the error could've been with the segment itself, or just an innocent mistake?

Reply:
Yes, exactly - we don't want to stop working with the transcoder if the response is "clean", even if that response indicates the transcoder encountered an error while transcoding. We only stop working if it seems that there's a connectivity issue with the transcoder.

There might well still be certain types of errors from the transcoder that would warrant dropping the T entirely, but we'd need to to take the time to analyze what those cases might be.

Reply:
I see. I agree that we should analyze what kind of errors from the transcoder itself would warrant dropping the T entirely. I imagine those cases do exist, but sounds like an optimization for later.